src/Microsoft.Azure.ServiceBus/Amqp/AmqpSubscriptionClient.cs

```diff
@@ -81,10 +81,7 @@ protected override Task OnCloseAsync()
                 amqpRequestMessage.Map[ManagementConstants.Properties.RuleDescription] =
                     AmqpMessageConverter.GetRuleDescriptionMap(description);
 
-                AmqpResponseMessage response =
-                    await
-                        ((AmqpMessageReceiver)this.InnerReceiver).ExecuteRequestResponseAsync(amqpRequestMessage)
-                            .ConfigureAwait(false);
+                await ((AmqpMessageReceiver)this.InnerReceiver).ExecuteRequestResponseAsync(amqpRequestMessage).ConfigureAwait(false);
```

Comment:
>await [](start = 16, length = 5)

Same here. we should handle Not OK case and throw with the correct exception here